Objective-C/CBLDocument.mm

```diff
@@ -173,8 +169,8 @@ - (BOOL) isEmpty {
 
 - (void) updateDictionary {
     if (_fleeceData) {
-        _root.reset(new MRoot<id>(new cbl::DocContext(_database, _c4Doc), Dict(_fleeceData), self.isMutable));
-        [_database safeBlock:^{
+        _root.reset(new MRoot<id>(new cbl::DocContext(_collection, _c4Doc), Dict(_fleeceData), self.isMutable));
```

Comment:
I'm not sure if MRoot need collection or database. Please trace the code to see where database / collection is used.

Reply:
It's used in CBLArray, CBLDictionary to get the '_db' lock. So it's not necessary to make this change and use collection.